CONTINUOUS_INTEGRATION.md

```diff
@@ -0,0 +1,198 @@
+ 4 -rwxrwxr-x  1 root root   538 Apr 20 22:52 host-build.sh
+ 4 -rwxrwxr-x  1 root root   556 Apr 20 22:52 node-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   343 Apr 20 22:52 node-tests.sh
+ 4 -rwxrwxr-x  1 root root   591 Apr 20 22:52 node_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   482 Apr 20 22:52 node_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root   614 Apr 20 22:52 package.sh
+ 4 -rwxrwxr-x  1 root root  2787 Apr 20 22:52 package-utils.sh
+ 4 -rwxrwxr-x  1 root root   521 Apr 20 22:52 python-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   274 Apr 20 22:52 python-tests.sh
+ 4 -rwxrwxr-x  1 root root   502 Apr 20 22:52 python_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   413 Apr 20 22:52 python_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root  2688 Apr 20 22:52 tf-build.sh
+ 4 -rwxrwxr-x  1 root root  2384 Apr 20 22:52 tf-package.sh
+ 4 -rwxrwxr-x  1 root root  3371 Apr 20 22:52 tf-setup.sh
+12 -rwxrwxr-x  1 root root  8375 Apr 20 22:52 tf-vars.sh
+```
+
+### Runners
+
+A _Runner_ is a server that listens for available _Jobs_ to run. The _Runner_ runs each _Job_ one at a time and then reports the results back to GitHub. A _Runner_ can be installed on a cloud server, or you can use the _Runner_ hosted by GitHub.
+
+If you are building DeepSpeech on particular hardware, or a non-Ubuntu variant of Linux (such as Arch), then you will need to [host your own _Runner_](https://docs.github.com/en/actions/hosting-your-own-runners).
+
+## Running continuous integration from GitHub
+
+In this example, we're going to assume that you will use a _Runner_ hosted on GitHub to perform continuous integration.
+
+This example assumes you have cloned DeepSpeech to the location:
+
+`https://github.com/{USERNAME}/DeepSpeech/`
+
+1. Navigate to your GitHub repository's Settings -> Actions. This should be located at `https://github.com/{USERNAME}/DeepSpeech/settings/actions`. It's a good idea for security to change the default Action Permissions from "Allow all actions" to "Allow local actions only".
+
+2. Next, navigate to your GitHub repository's Actions. This should be located at `https://github.com/{USERNAME}/DeepSpeech/actions`. You should see three _Workflows_ already set up. GitHub recognises these from the files stored in the `.github/workflows` directory. The _Workflows_ are set up to execute on a `push` event.
+
+3. So, to test that they are working, we need to make a change to the `DeepSpeech` repository to trigger a `push` event. Navigate to the repository's Code listing. This should be at `https://github.com/{USERNAME}/DeepSpeech`. Add a file called `test-workflow.txt` containing dummy text and commit the change. This should trigger a `push` event, and the _Workflows_ will activate. You can monitor the _Workflows_ executing from the repository's Actions page. This is at `https://github.com/{USERNAME}/DeepSpeech/actions`. The _Workflow_ takes around 4hrs 15mins to run using the built-in GitHub Runner.
```

Comment:
Running for 4h seems NOT normal. Completion should not take more than a bit more than one hour.

Reply:
Sure -  https://github.com/KathyReid/DeepSpeech/actions
For now I will remove the estimated completion time until we have a clearer idea of how long it takes based on multiple samples.

Reply:
I see you only had one run, tensorflow cache first population would account, if you run à second time a new pr it should be faster

Reply:
Confirming that it was much faster on the second run - only an hour or so
https://github.com/KathyReid/DeepSpeech/actions